Index: octomap-v1.4.2/octomap/CMakeModules/CompilerSettings.cmake
===================================================================
--- octomap-v1.4.2/octomap/CMakeModules/CompilerSettings.cmake	(revision 447)
+++ octomap-v1.4.2/octomap/CMakeModules/CompilerSettings.cmake	(working copy)
@@ -31,10 +31,25 @@
 
 # Set full rpath http://www.paraview.org/Wiki/CMake_RPATH_handling
 # (good to have and required with ROS)
-set(CMAKE_SKIP_BUILD_RPATH  FALSE)
-set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
+if(UNIX)
+    target_link_libraries(${qhull_SHARED} m)
+    if(APPLE)
+        set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+    else()
+        set(CMAKE_SKIP_BUILD_RPATH  FALSE)
+        set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+        set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+        set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+    endif()
+endif(UNIX)
+
+
+
+#set(CMAKE_SKIP_BUILD_RPATH  FALSE)
+#set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+#set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+#set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
 # no prefix needed for python modules
 set(CMAKE_SHARED_MODULE_PREFIX "")
